source/developer/interactive-messages.rst

```diff
@@ -0,0 +1,329 @@
+
+.. image:: ../../source/images/matterpoll.png
+
+.. toctree::
+  :maxdepth: 2
+
+Message Buttons
+----------------
+
+Add message buttons as ``actions`` in your integration `message attachments <https://docs.mattermost.com/developer/message-attachments.html>`_.
+
+.. image:: ../../source/images/poll.gif
+
+The following payload gives an example that uses message buttons.
+
+.. code-block:: text
+
+  {
+    "attachments": [
+      {
+        "pretext": "This is the attachment pretext.",
+        "text": "This is the attachment text.",
+        "actions": [
+          {
+            "name": "Ephemeral Message",
+            "integration": {
+              "url": "http://127.0.0.1:7357",
+              "context": {
+                "action": "do_something_ephemeral"
+              }
+            }
+          }, {
+            "name": "Update",
+            "integration": {
+              "url": "http://127.0.0.1:7357",
+              "context": {
+                "action": "do_something_update"
+              }
+            }
+          }
+        ]
+      }
+    ]
+  }
+
+The integration can respond with an update to the original post, or with an ephemeral message:
+
+.. code-block:: text
+
+  {
+    "update": {
+      "message": "Updated!"
+    },
+    "ephemeral_text": "You updated the post!"
+  }
+
+.. image:: ../../source/images/interactive_message.gif
+
+Message Menus
+----------------
+
+Similar to buttons, add message menus as ``actions`` in your integration `message attachments <https://docs.mattermost.com/developer/message-attachments.html>`_.
+
+.. image:: ../../source/images/message_menus.png
+
+The following payload gives an example that uses message menus.
+
+.. code-block:: text
+
+  {
+    "attachments": [
+      {
+        "pretext": "This is the attachment pretext.",
+        "text": "This is the attachment text.",
+        "actions": [
+          {
+            "name": "Select an option...",
+            "integration": {
+              "url": "http://127.0.0.1:7357/action_options",
+              "context": {
+                "action": "do_something"
+              }
+            },
+            "type": "select",
+            "options": [
+                    {
+                        "text": "Option1",
+                        "value": "opt1"
+                    },
+                    {
+                        "text": "Option2",
+                        "value": "opt2"
+                    },
+                    {
+                        "text": "Option3",
+                        "value": "opt3"
+                    }
+            ]
+          }
+        ]
+      }
+    ]
+  }
+
+The integration can respond with an update to the original post, or with an ephemeral message:
+
+.. code-block:: text
+
+  {
+    "update": {
+      "message": "Updated!"
+    },
+    "ephemeral_text": "You updated the post!"
+  }
+
+.. image:: ../../source/images/message-menu.png
```

Comment:
Image doesn't render, (or maybe this isn't supposed to be here since it looks like it's the same image as above?)